Modules/+Experiments/+AutoExperiment/@SpecSlowScan/SpecSlowScan.m

```diff
@@ -171,33 +174,36 @@ function PreSlow(obj,slow_experiment)
         function sites = AcquireSites(obj,managers)
             sites = Experiments.AutoExperiment.AutoExperiment_invisible.SiteFinder_Confocal(managers,obj.imaging_source,obj.site_selection);
         end
-        function PreRun(obj,status,managers,ax)
-            if ~isempty(obj.analysis)
-                status.String = 'Checking analysis file'; drawnow;
-                % We already checked size(...,2) in validate_file; at this
-                % point there should be data loaded as well!
-                n_analysis_sites = size(obj.analysis,1);
-                n_data_sites = length(obj.data.sites);
-                assert(n_analysis_sites==n_data_sites,...
-                    sprintf('Found %i analysis entries, but %i sites. These should be equal.',...
-                    	n_analysis_sites,n_data_sites));
-                for i = 1:n_data_sites
-                    for j = 1:3
-                        assert(isnan(obj.analysis(i,j).index) || (obj.analysis(i,j).index == i),...
-                            ['At least one analysis index does not reference its position (also corresponding to data position). ',...
-                            'This is currently not supported and likely means the "inds" option was used in the analysis method.']);
-                    end
+        function validate_analysis(obj)
+            for i = 1:size(obj.analysis.sites,1)
+                for j = 1:3 % The 3 is for n experiments, which was validated in superclass
+                    assert(isnan(obj.analysis.sites(i,j).index) || (obj.analysis.sites(i,j).index == i),...
+                        ['At least one analysis index does not reference its position (also corresponding to data position). ',...
+                        'This is currently not supported and likely means the "inds" option was used in the analysis method.']);
                 end
             end
+        end
+        function PreRun(obj,status,managers,ax)
             %before running, calibrate spectrometer and check resLaser
             status.String = 'Checking spectrometer and resLaser'; drawnow;
+            assert(~isempty(obj.experiments(2).repumpLaser),'SlowScan.Open needs a repump laser defined!');
```

Comment:
Should these checks live in the experiments themselves? I.e. should ALL experiments have a prerun method?

Reply:
Yep - we've talked about including a `check_ready` or something for experiments, but until someone does that they need to live here.